driver/ppm_events.c

```diff
@@ -603,20 +613,24 @@ int val_to_ring(struct event_filler_arguments *args, uint64_t val, u32 val_len,
 	case PT_FSPATH:
 	case PT_FSRELPATH:
 		if (likely(val != 0)) {
+#ifndef WDIG
 			if (fromuser) {
+#endif
 				len = ppm_strncpy_from_user(args->buffer + args->arg_data_offset,
-					(const char __user *)(unsigned long)val, max_arg_size);
+					(const char __user *)(syscall_arg_t)val, max_arg_size);
 
 				if (unlikely(len < 0))
 					return PPM_FAILURE_INVALID_USER_MEMORY;
+#ifndef WDIG
 			} else {
 				len = strlcpy(args->buffer + args->arg_data_offset,
-								(const char *)(unsigned long)val,
+								(const char *)(syscall_arg_t)val,
 								max_arg_size);
 
 				if (++len > max_arg_size)
 					len = max_arg_size;
 			}
+#endif
```

Comment:
```
#ifdef WDIG
fromuser = 1; // because XXX?
#endif 
```

and drop the ifndefs? (I probably got the `fromuser = 1` part wrong)

Reply:
I can potentially drop the first ifdef, but I cannot do what you suggest because strlcpy does not exist on windows.

I added a comment to explain what the ifdef does.

Reply:
actually, I provided an implementation of strlcpy under windows and then I fixed this as suggested